stack/services/src/main/java/org/apache/usergrid/services/assets/data/S3BinaryStore.java

```diff
@@ -134,10 +134,11 @@ public void write( final UUID appId, final Entity entity, InputStream inputStrea
         }
         else { // bigger than 5mb... dump 5 mb tmp files and upload from them
 
-            // todo: yes, AsyncBlobStore is deprecated, but there appears to be no replacement yet
-            final AsyncBlobStore blobStore = getContext().getAsyncBlobStore();
+            // create temp file and copy entire file to that temp file
 
-            File tempFile = File.createTempFile( entity.getUuid().toString(), "tmp" );
+            LOG.debug( "Writing temp file for S3 upload" );
```

Comment:
This code itself looks great.  However I noticed on the total impl we don't have a max file size we support.  Since we're going to have to upload this to the tomcat server, should we enforce a max file size to stop someone from overloading the tomcats?